test/core/event_engine/test_suite/event_engine_test_utils.cc

```diff
@@ -44,6 +45,45 @@ using Listener = ::grpc_event_engine::experimental::EventEngine::Listener;
 namespace grpc_event_engine {
 namespace experimental {
 
+namespace {
+
+constexpr int kMinMessageSize = 1024;
+constexpr int kMaxMessageSize = 4096;
+
+}  // namespace
+
+// Returns a random message with bounded length.
+std::string GetNextSendMessage() {
+  static const char alphanum[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+  static std::random_device rd;
+  static std::seed_seq seed{rd()};
+  static std::mt19937 gen(seed);
+  static std::uniform_real_distribution<> dis(kMinMessageSize, kMaxMessageSize);
+  static grpc_core::Mutex g_mu;
+  std::string tmp_s;
+  int len;
+  {
+    grpc_core::MutexLock lock(&g_mu);
+    len = dis(gen);
+  }
+  tmp_s.reserve(len);
+  for (int i = 0; i < len; ++i) {
+    tmp_s += alphanum[rand() % (sizeof(alphanum) - 1)];
+  }
+  return tmp_s;
+}
+
+// Waits until the use_count of the event engine shared_ptr has reached 1
+// and returns.
+void WaitForPendingTasks(std::shared_ptr<EventEngine>&& engine) {
```

Comment:
This presumes that all tasks own a shared_ptr to this EventEngine, which even in this PR is not the case. I'd suggest a rename to `WaitForSingleOwner` or some such, with the caveat that some callbacks _may still be running_ when this reaches its condition.

Reply:
Renamed to WaitForSingleOwner